lib/Webhooks.js

```diff
@@ -1,277 +1,174 @@
+        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
+        step((generator = generator.apply(thisArg, _arguments || [])).next());
+    });
+};
 const utils = require('./utils');
-const {StripeError, StripeSignatureVerificationError} = require('./Error');
-
+const { StripeError, StripeSignatureVerificationError } = require('./Error');
 const Webhook = {
-  DEFAULT_TOLERANCE: 300, // 5 minutes
-
-  constructEvent(payload, header, secret, tolerance, cryptoProvider) {
-    this.signature.verifyHeader(
-      payload,
-      header,
-      secret,
-      tolerance || Webhook.DEFAULT_TOLERANCE,
-      cryptoProvider
-    );
-
-    const jsonPayload = JSON.parse(payload);
-    return jsonPayload;
-  },
-
-  async constructEventAsync(
-    payload,
-    header,
-    secret,
-    tolerance,
-    cryptoProvider
-  ) {
-    await this.signature.verifyHeaderAsync(
-      payload,
-      header,
-      secret,
-      tolerance || Webhook.DEFAULT_TOLERANCE,
-      cryptoProvider
-    );
-
-    const jsonPayload = JSON.parse(payload);
-    return jsonPayload;
-  },
-
-  /**
-   * Generates a header to be used for webhook mocking
-   *
-   * @typedef {object} opts
-   * @property {number} timestamp - Timestamp of the header. Defaults to Date.now()
-   * @property {string} payload - JSON stringified payload object, containing the 'id' and 'object' parameters
-   * @property {string} secret - Stripe webhook secret 'whsec_...'
-   * @property {string} scheme - Version of API to hit. Defaults to 'v1'.
-   * @property {string} signature - Computed webhook signature
-   * @property {CryptoProvider} cryptoProvider - Crypto provider to use for computing the signature if none was provided. Defaults to NodeCryptoProvider.
-   */
-  generateTestHeaderString: function(opts) {
-    if (!opts) {
-      throw new StripeError({
-        message: 'Options are required',
-      });
-    }
-
-    opts.timestamp =
-      Math.floor(opts.timestamp) || Math.floor(Date.now() / 1000);
-    opts.scheme = opts.scheme || signature.EXPECTED_SCHEME;
-
-    opts.cryptoProvider = opts.cryptoProvider || getNodeCryptoProvider();
-
-    opts.signature =
-      opts.signature ||
-      opts.cryptoProvider.computeHMACSignature(
-        opts.timestamp + '.' + opts.payload,
-        opts.secret
-      );
-
-    const generatedHeader = [
-      't=' + opts.timestamp,
-      opts.scheme + '=' + opts.signature,
-    ].join(',');
-
-    return generatedHeader;
-  },
+    DEFAULT_TOLERANCE: 300,
+    constructEvent(payload, header, secret, tolerance, cryptoProvider) {
+        this.signature.verifyHeader(payload, header, secret, tolerance || Webhook.DEFAULT_TOLERANCE, cryptoProvider);
+        const jsonPayload = JSON.parse(payload);
+        return jsonPayload;
+    },
+    constructEventAsync(payload, header, secret, tolerance, cryptoProvider) {
```

Comment:
I think we may want to bump the spec to ES2017 to not include that __awaiter helper method if we have already used async/await.

Reply:
Everything we use from ES2017 is supported by Node 8 so I think the change is pretty safe: https://node.green/#ES2017